### PR TITLE
[Compiler-v2] Fix parsing of `location` in plan_builder of compiler v2

### LINKED ISSUE #14635: [Bug][move-compiler-v2] #[expected_failure] with invalid location yields no error in V2
# 🐛 Bug

See test output
- third_party/move/move-compiler/tests/move_check/unit_test/other_failures_invalid_location.unit_test.exp
- third_party/move/move-compiler-v2/tests/unit_test/test/other_failures_invalid_location.exp
Note that V1 shows errors for lines 26, 30, 34, 38, such as:
```
error[E10003]: invalid attribute value
   ┌─ tests/move_check/unit_test/other_failures_invalid_location.move:26:38
   │
26 │     #[expected_failure(vector_error, location=x)]
   │                                      ^^^^^^^^^^
   │                                      │        │
   │                                      │        Expected a module identifier, e.g. 'std::vector'
   │                                      Invalid value in attribute assignment
```
Compiler V2 misses these errors.

### PR DESCRIPTION
**Pull request by rahxephon89**
## Description

This PR closes #14635 by fixing how `location` is parsed in the attribute for move unit tests.

## How Has This Been Tested?

Existing tests.

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [x] Move Compiler
- [ ] Other (specify)

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: third_party/move/move-compiler-v2/src/plan_builder.rs [hunk 1]
```diff
@@ -505,10 +505,17 @@ fn convert_location(env: &GlobalEnv, attr: Attribute) -> Option<ModuleId> {
     match value {
         AttributeValue::Name(id, opt_module_name, _sym) => {
             let vloc = env.get_node_loc(id);
-            convert_module_id(env, vloc, opt_module_name)
+            let module_id_opt = convert_module_id(env, vloc.clone(), opt_module_name);
+            if !_sym.display(env.symbol_pool()).to_string().is_empty() || module_id_opt.is_none() {
```
Comment (vineethk) [inline third_party/move/move-compiler-v2/src/plan_builder.rs:509]:
```suggestion
            if !sym.display(env.symbol_pool()).to_string().is_empty() || module_id_opt.is_none() {
```

Comment (vineethk) [reply third_party/move/move-compiler-v2/src/plan_builder.rs:509]: Also change `_sym` -> `sym` elsewhere.

Comment (vineethk) [reply third_party/move/move-compiler-v2/src/plan_builder.rs:509]: Is there a test case corresponding to `!_sym.display(env.symbol_pool()).to_string().is_empty()`?

File: third_party/move/move-compiler-v2/src/plan_builder.rs [hunk 2]
```diff
@@ -505,10 +505,17 @@ fn convert_location(env: &GlobalEnv, attr: Attribute) -> Option<ModuleId> {
...
+                env.error_with_labels(&loc, "invalid attribute value", vec![(
+                    vloc,
+                    "Expected a module identifier, e.g. 'std::vector'".to_string(),
+                )]);
+            }
+            module_id_opt
         },
         AttributeValue::Value(id, _val) => {
-            let vloc = env.get_node_loc(id);
+            let vloc: Loc = env.get_node_loc(id);
```
Comment (vineethk) [inline third_party/move/move-compiler-v2/src/plan_builder.rs:518]: Was this explicit type annotation temporarily added for debugging?

**Review by vineethk — APPROVED** [review]
Approving assuming comments are addressed.

File: third_party/move/move-compiler-v2/src/plan_builder.rs (hunk 1 above)
Comment (rahxephon89) [reply third_party/move/move-compiler-v2/src/plan_builder.rs:509]: yes, in the test case where we have `#[expected_failure(out_of_gas, location=0x1::m::t0)]` and `_sym` corresponds to `t0`.

Comment (vineethk) [reply third_party/move/move-compiler-v2/src/plan_builder.rs:509]: And when would it be an empty string?

**Comment by rahxephon89** [comment]
* **#15312** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/15312?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/15312?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: third_party/move/move-compiler-v2/src/plan_builder.rs (hunk 1 above)
Comment (rahxephon89) [reply third_party/move/move-compiler-v2/src/plan_builder.rs]: when it is a module: `0x1::m`. Added a test for this case.

File: third_party/move/move-compiler-v2/src/plan_builder.rs (hunk 2 above)
Comment (rahxephon89) [reply third_party/move/move-compiler-v2/src/plan_builder.rs]: done

**Review by brmataptos — APPROVED** [review]
It would be nice to better localize the errors, but this is OK.

**Pull request merged**